clippy_lints/src/utils/internal_lints.rs

```diff
@@ -207,3 +218,26 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for LintCollector<'a, 'tcx> {
         NestedVisitorMap::All(&self.cx.tcx.hir)
     }
 }
+
+
+
+pub struct DefaultHashTypes;
+
+impl LintPass for DefaultHashTypes {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(DEFAULT_HASH_TYPES)
+    }
+}
+
+impl EarlyLintPass for DefaultHashTypes {
+    fn check_ident(&mut self, cx: &EarlyContext<'_>, ident: Ident) {
+        let mut map = HashMap::default();
```

Comment:
You should do this in a `new()` function on creating the Lint. Doing this here will create a new `HashMap` everytime you check an `ident`. This happens pretty often and is bad for the performance.

Reply:
Oh and you're using `HashMap` instead of `FxHashMap`. :smile: